AppCenterData/AppCenterData/MSPaginatedDocuments.m

```diff
@@ -16,22 +16,29 @@ @implementation MSPaginatedDocuments
 @synthesize continuationToken = _continuationToken;
 @synthesize partition = _partition;
 @synthesize documentType = _documentType;
+@synthesize deviceTimeToLive = _deviceTimeToLive;
 
 - (instancetype)initWithPage:(MSPage *)page
                    partition:(NSString *)partition
                 documentType:(Class)documentType
+            deviceTimeToLive:(NSInteger)deviceTimeToLive
            continuationToken:(NSString *_Nullable)continuationToken {
   if ((self = [super init])) {
     _currentPage = page;
     _partition = partition;
     _documentType = documentType;
+    _deviceTimeToLive = deviceTimeToLive;
     _continuationToken = continuationToken;
   }
   return self;
 }
 
 - (instancetype)initWithError:(MSDataError *)error partition:(NSString *)partition documentType:(Class)documentType {
-  return [self initWithPage:[[MSPage alloc] initWithError:error] partition:partition documentType:documentType continuationToken:nil];
+  return [self initWithPage:[[MSPage alloc] initWithError:error]
+                  partition:partition
+               documentType:documentType
+           deviceTimeToLive:0
```

Comment:
Can you check what value we return on Android?

I would almost introduce a dedicated constant for "unknown" device time to live value instead. Or at least use the `kMSDataTimeToLiveNoCache` constant.

Reply:
Will use kMSDataTimeToLiveNoCache. Android doesnt seem to have an initWithError kind of implementation

Reply:
That's because passing the error to the customer is different in Android. The error/exception pattern are not the same between platforms.